wallets/provider-walletconnect-2/src/helpers.ts

```diff
@@ -282,3 +207,34 @@ export function getEvmAccount(
     address,
   });
 }
+
+// It's enough to return only connected network for the EVM networks and ignore others
+export function filterEvmAccounts(
```

Comment:
Maybe i'm wrong and I didn't test, but does this works too?

```
export function findFirstEvmAccount(accounts, currentChainId) {
  const account = accounts.find((account) => {
    const { address, chainId } = account;
    return isEvmAddress(address) && chainId == currentChainId;
  });

  return account;
}
```

Reply:
It's not the same logic. Firstly, I don't want to filter non-evm accounts but this code only returns evm account. Secondly, if the current chain id is undefined, I return the first evm account and exclude other evm accounts.
But I update it to make it more readable.